state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1156,131 +1194,202 @@ pub mod pallet {
+					swap.stable_amount.unwrap_or_else(|| {
+						log_or_panic!("stable amount must be set for swaps from STABLE_ASSET");
 						swap.input_amount()
-					};
+					})
+				} else {
+					swap.input_amount()
+				};
 
-					Self::deposit_event(Event::<T>::SwapExecuted {
-						swap_id: swap.swap_id(),
-						source_asset: swap.input_asset(),
-						destination_asset: swap.output_asset(),
-						deposit_amount: swap_input,
-						swap_input,
-						egress_amount: swap_output,
-						swap_output,
-						intermediate_amount: swap.intermediate_amount(),
-						swap_type: swap.swap_type().clone(),
-					});
-					// Handle swap completion logic.
-					match &swap.swap_type() {
-						SwapType::Swap(destination_address) =>
-							match T::EgressHandler::schedule_egress(
-								swap.output_asset(),
-								swap_output,
-								destination_address.clone(),
-								None,
-							) {
-								Ok(ScheduledEgressDetails {
-									egress_id,
-									egress_amount,
-									fee_withheld,
-								}) => {
-									Self::deposit_event(Event::<T>::SwapEgressScheduled {
-										swap_id: swap.swap_id(),
-										egress_id,
-										asset: swap.output_asset(),
-										amount: egress_amount,
-										fee: fee_withheld,
-									});
-								},
-								Err(err) => {
-									Self::deposit_event(Event::<T>::SwapEgressIgnored {
-										swap_id: swap.swap_id(),
-										asset: swap.output_asset(),
-										amount: swap_output,
-										reason: err.into(),
-									});
-								},
+				let swap_request_id = swap.swap.swap_request_id;
+
+				let Some(mut request) = SwapRequests::<T>::take(swap_request_id) else {
+					log_or_panic!("swap request not found");
+					continue;
+				};
+
+				let Some(output_amount) = swap.final_output else {
+					log_or_panic!("Swap is not completed yet!");
+					continue;
+				};
+
+				Self::deposit_event(Event::<T>::SwapExecuted {
+					swap_request_id,
+					swap_id: swap.swap_id(),
+					input_amount: swap_input,
+					input_asset: swap.input_asset(),
+					network_fee: swap.network_fee_taken.unwrap_or_default(),
+					output_asset: swap.output_asset(),
+					output_amount,
+					intermediate_amount: swap.intermediate_amount(),
+				});
+
+				match &mut request.state {
+					SwapRequestState::Ccm { state, ccm_deposit_metadata, output_address } =>
+						match state {
+							CcmSwapState::PrincipalSwapScheduled {
+								principal_swap_id,
+								gas_budget,
+								other_gas_asset,
+							} => {
+								debug_assert!(swap.swap_id() == *principal_swap_id);
+
+								if let Some(other_gas_asset) = other_gas_asset {
+									*state = Self::schedule_ccm_gas_swap(
+										swap_request_id,
+										request.input_asset,
+										*other_gas_asset,
+										*gas_budget,
+										output_amount,
+									);
+
+									// We are not done with the request, so put it back
+									SwapRequests::<T>::insert(swap_request_id, request);
+								} else {
+									Self::process_ccm_outcome(
```

Comment:
should we Self::deposit_event(Event::<T>::SwapRequestCompleted { swap_request_id }); here too?

Reply:
As, above this is done in `process_ccm_outcome`.